src/main/java/org/wso2/carbon/module/swiftiso20022/validation/rules/custom/MT940TransactionTypeValidationRule.java

```diff
@@ -0,0 +1,90 @@
+ */
+
+package org.wso2.carbon.module.swiftiso20022.validation.rules.custom;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONObject;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+import org.wso2.carbon.module.swiftiso20022.constants.MT940Constants;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidationResult;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidationRule;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidatorContext;
+
+import java.util.List;
+
+/**
+ * MT940 Transaction Type Validation Rule.
+ */
+public class MT940TransactionTypeValidationRule extends ValidationRule {
+
+    ValidatorContext context;
+    private static final String RULE_NAME = "MT940 Transaction Type Validation";
+    private static final int INDEX = 1;
+    private static final int START_VALUE = 99;
+    private static final int END_VALUE = 1000;
+
+    private static final List<String> supportedFields = List.of(MT940Constants.SWIFT_TRANSFER,
+            MT940Constants.NON_SWIFT_TRANSFER, MT940Constants.FIRST_ADVICE);
+
+    public MT940TransactionTypeValidationRule(ValidatorContext context) {
+        this.context = context;
+    }
+
+    /**
+     * Validate whether the parameter Transaction type in MT940 is valid.
+     * @return Validation Result
+     */
+    @Override
+    public ValidationResult validate(JSONObject payload) {
+        if (payload.has(context.getFieldName())) {
+            Object transactionType = payload.get(context.getFieldName());
+
+            if (!(transactionType instanceof String) || StringUtils.isBlank(transactionType.toString())) {
+                return new ValidationResult(ConnectorConstants.ERROR_T53,
+                        String.format(ConnectorConstants.ERROR_PARAMETER_INVALID,
+                                context.getFieldDisplayName()));
+            }
+            boolean isSupported = supportedFields.stream().anyMatch(transactionType.toString()::startsWith);
+            if (!isSupported) {
+                return new ValidationResult(ConnectorConstants.ERROR_T53,
+                        ConnectorConstants.ERROR_TRANS_TYPE_INVALID);
+            }
+
+            if (transactionType.toString().startsWith(MT940Constants.SWIFT_TRANSFER)) {
+                String identificationCode = transactionType.toString().substring(INDEX);
+                if (!StringUtils.isNumeric(identificationCode)) {
+                    return new ValidationResult(ConnectorConstants.ERROR_T53,
+                            ConnectorConstants.ERROR_TRANS_TYPE_INVALID);
+                }
+
+                if (!(Integer.parseInt(identificationCode) > START_VALUE &&
+                        Integer.parseInt(identificationCode) < END_VALUE)) {
+                    return new ValidationResult(ConnectorConstants.ERROR_T18,
+                            ConnectorConstants.ERROR_TRANS_TYPE_INVALID);
+                }
+            }
```

Comment:
Can't we move this logic to a utility class since it is similar to the MT940StatementTypeValidationRule ?

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/6/commits/7f2058c938da71ec80755d296fd2d3d67541e678